### PR TITLE
feature(TerraForm): validation for form field with min/max

### PR DESCRIPTION
**Pull request by niwolf**
@plentymarkets/team-terra

#### Must be done by Terra

Testing

-   [ ] Person 1 (mandatory)

    Browser support (relevant for changes in scss / appearance of components)

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

-   [ ] Person 2 (mandatory)

    Browser support

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

Information transfer

-   [ ] Inform Ceres about changes in `TerraFormComponent` (optional: relevant for changes in this component)

Terra Basic Plugin

-   [ ] Adapt changes (optional: relevant for version updates, global design changes, l10n)

---

#### Must be done by every developer

Please inform a member of Terra (@plentymarkets/team-terra) to get the upper part of the checklist done (with urgency or deadline).

Documentation

-   [ ] Changelog (optional: relevant for every change which influences the API)
-   [ ] Example (updated or created)
-   [ ] JSDoc (optional: relevant for every change which influences the API)
-   [ ] [Google spreadsheet](https://docs.google.com/spreadsheets/d/1OINnux8TEoitV-qdAxqUQaf9oGI_fqFVwfRWenRFfhI/edit#gid=0) (relevant when deprecating public APIs or features)

Testing

-   [ ] Unit Test (updated or created)

### CODE REVIEW

File: src/lib/components/forms/form/form-components/slider/slider.component.html [hunk 1]
```diff
@@ -22,3 +22,9 @@
 <mat-error *ngIf="isRequired && control.hasError('required')">
     {{ 'validators.required' | translate: _locale.language }}
 </mat-error>
+<mat-error *ngIf="min && control.hasError('min')">
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/slider/slider.component.html:25]: Gefährlich.. min kann halt auch 0 sein.. müsste stattdessen also überprüft werden ob es nicht null oder undefined ist.
Oder du lässt das komplett aus der Bedingung raus. Der zweite Teil trifft ja eh nur zu wenn der validator auch gesetzt ist.

```suggestion
<mat-error *ngIf="control.hasError('min')">
```

File: src/lib/components/forms/form/form-components/slider/slider.component.html [hunk 2]
```diff
@@ -22,3 +22,9 @@
...
+    {{ 'validators.min' | translate: _locale.language }}
+</mat-error>
+<mat-error *ngIf="max && control.hasError('max')">
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/slider/slider.component.html:28]: Siehe oben

File: src/lib/components/forms/form/form-components/number-input/number-input.component.html [hunk 3]
```diff
@@ -17,4 +17,10 @@
     <mat-error *ngIf="isRequired && control.hasError('required')">
         {{ 'validators.required' | translate: _locale.language }}
     </mat-error>
+    <mat-error *ngIf="minValue && control.hasError('min')">
```
[the remark this answers is not in the document]
Comment (pweyrich) [reply src/lib/components/forms/form/form-components/number-input/number-input.component.html:20]: habe nochmal geschaut. Es ist tatsächlich `minValue` und `maxValue`

[the remark this answers is not in the document]
Comment (niwolf) [reply src/lib/components/forms/form/form-components/number-input/number-input.component.html:20]: Du meinst also - es sollte heißen `control.hasError('minValue')">` ?

[the remark this answers is not in the document]
Comment (pweyrich) [reply src/lib/components/forms/form/form-components/number-input/number-input.component.html:20]: Nope. Die Inputs müssen sowohl beim NumberInput als auch beim Slider minValue und maxValue heißen.

Comment (pweyrich) [inline src/lib/components/forms/form/form-components/number-input/number-input.component.html:20]:
```suggestion
    <mat-error *ngIf="control.hasError('min')">
```

File: src/lib/components/forms/form/form-components/number-input/number-input.component.html [hunk 4]
```diff
@@ -17,4 +17,10 @@
...
+        {{ 'validators.min' | translate: _locale.language }}
+    </mat-error>
+    <mat-error *ngIf="maxValue && control.hasError('max')">
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/number-input/number-input.component.html:23]:
```suggestion
    <mat-error *ngIf="control.hasError('max')">
```

**Review by nappynapster — APPROVED** [review]

**Comment by niwolf** [comment]
postponed after the release of Angular v12

**Review by Adonsio — APPROVED** [review]

**Review by nappynapster — DISMISSED** [review]

**Review by Adonsio — DISMISSED** [review]

**Review by pweyrich — APPROVED** [review]
Tested ✅
Code Review ✅  
Although error messages for the slider won't ever show up.

**Review by pweyrich — APPROVED** [review]

**Review by Adonsio — APPROVED** [review]

**Review by nappynapster — APPROVED** [review]

**Pull request merged**